### DISCUSSION TITLE
Is there a way to quickly browse to sub-orchestrations?

### DISCUSSION THREAD

**Discussion by StephenAstrumU** [microsoft/DurableFunctionsMonitor#126] [Q&A] [answered]
When viewing logs for an orchestration, I can see the sub-orchestrations initiated for an orchestrator. Is there a way to quickly browse into the sub-orchestrator (e.g. to see what activities or sub-orchestrations that sub-orchestrator spawned)? The timestamp is clickable but only modifies the time filter of the current orchestration.

We're running the durable function monitor as a Nuget package injected into an azure function if it matters

**Comment by scale-tone** [comment]
Hi @StephenAstrumU ,

> Is there a way to quickly browse into the sub-orchestrator

Certainly. 

#### Option1. Clickable links to subinstances should appear in the parent instance's history:
![image](https://github.com/microsoft/DurableFunctionsMonitor/assets/5447190/5f8547d5-cb02-45cf-9bfc-dd1d43be7c95)

#### Option2. Both 'Sequence Diagram' and 'Gantt Chart' tabs show the whole picture, including subinstances:
![image](https://github.com/microsoft/DurableFunctionsMonitor/assets/5447190/8cd73128-b534-4bc6-ba54-ad795ee32440)

#### Option3. You can filter by `parentInstanceId` field:
![image](https://github.com/microsoft/DurableFunctionsMonitor/assets/5447190/a8995265-0646-4bd7-a400-7e7d392c8673)

**Reply by StephenAstrumU** [reply]
Thanks for the reply!

There must be more going on with our monitor set-up, then. The name in the table is not clickable and our Gantt Chart and Sequence Diagram only show the "first level" of child instances (instances kicked off by child instances analogous to `TinoSubSubOrchestrator` are not visible). 

I am seeing a warning "Failed to get execution history from storage, falling back to DurableClient". So there's probably some other issue at play. FWIW we're using netherite

**Comment by scale-tone** [comment, marked as answer]
> FWIW we're using netherite

Yeah, that explains it. Sadly, netherite storage provider currently does not return child instance ids as part of execution history, and to my best knowledge, there isn't any known way of overriding that behavior.

Only two things I can suggest:
1. You could implement a custom history extraction routine and provide it to DfMon [via DfmExtensionPoints](https://github.com/microsoft/DurableFunctionsMonitor/blob/main/durablefunctionsmonitor.dotnetbackend/Common/Setup.cs#L97). If your solution allows for any custom way of correlating parent-child instances (e.g. matching them by instanceIds), you could implement it in that routine and return the history with [HistoryEvent.SubOrchestrationId](https://github.com/microsoft/DurableFunctionsMonitor/blob/main/durablefunctionsmonitor.dotnetbackend/Common/OrchestrationHistory.cs#L204) fields populated. Then DfMon will render it properly.
2. You could submit an issue to [durabletask-netherite repo](https://github.com/microsoft/durabletask-netherite/issues) regarding this. This is what I suggest you to do anyway.

**Reply by StephenAstrumU** [reply]
Glad to have a definitive answer at least.

Submitted https://github.com/microsoft/durabletask-netherite/issues/296

**Reply by sebastianburckhardt** [reply]
> Sadly, netherite storage provider currently does not return child instance ids as part of execution history

I am a bit surprised about that. Netherite does not interpret or modify execution history in any way.
Based on how the `SubOrchestrationInstanceCreated` history event is defined I would think the `InstanceId` field (https://github.com/Azure/durabletask/blob/ff6b1c264dce132d568e74196c2b159474f9aa08/src/DurableTask.Core/History/SubOrchestrationInstanceCreatedEvent.cs#L54) should contain the instance id of the suborchestration. Perhaps this field was renamed by the AzureStorageProvider to a different name so it now does not match?
